platform/pulp/app/tasks/publisher.py

```diff
@@ -15,3 +17,27 @@ def delete(repo_name, publisher_name):
     :type  publisher_name:  str
     """
     models.Publisher.objects.filter(name=publisher_name, repository__name=repo_name).delete()
+
+@shared_task(base=UserFacingTask)
+def publish(repo_name, publisher_name):
+    """
+    Call publish on the publisher defined by a plugin.
+
+    Check that the publisher has a repository, which is necessary to publish. A working directory
+    is prepared, the plugin's publish is called, and then working directory is removed.
+
+    Args:
+        repo_name (basestring): unique name to specify the repository.
```

Comment:
s/basestring/str/ 

https://docs.python.org/3.0/whatsnew/3.0.html#text-vs-data-instead-of-unicode-vs-8-bit